server/core/region.go

```diff
@@ -402,6 +548,18 @@ func (r *RegionsInfo) AddRegion(region *RegionInfo) []*metapb.Region {
 		store.Put(region)
 	}
 
+	for _, peer := range region.PendingPeers {
+		if !peer.IsLearner {
+			continue
+		}
+		storeID := peer.GetStoreId()
+		store, ok := r.pendingLearners[storeID]
+		if !ok {
+			store = newRegionMap()
+			r.pendingLearners[storeID] = store
+		}
+		store.Put(region)
```

Comment:
now, `max-pending-peer-count`  should care about `pending-peer` and `pending-learn`.